### PR TITLE
ci: add e2e tests with GH Actions and Kubernetes Kind

### LINKED ISSUE #13: Add e2e smoke test with Kubernetes Kind

### PR DESCRIPTION
**Pull request by stefanprodan**
Add e2e testing with Kubernetes Kind:
- generate SSH key secret
- build and load container image in Kind
- deploy gitsrv and mount SSH key
- wait for deployment to be ready
- generate known_hosts with ssh-keyscan
- upload known_hosts to GitHub Actions artifacts
- print debug info on deployment failure 

Fix: #13

### CODE REVIEW

File: test/known_hosts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# generated with "ssh-keyscan gitsrv"
+gitsrv ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC3qgBx5WLHbB+frq8d+iomxrdQPlWcj7NsJSiEz8vYOaRQsQDMtWHHYRt+zBVIH9Q96urBHDZa+B72DsInkO7JxflLVXLenwhNu8u9a0VbbleYnBYWfoNYWv+0tQpilOIasNjA+aLAWsVVjEAPB8RM77QocXnUbthI898f5EbQ1jXLBMPfbY0VRHimqo0EST71q0/PzfQTf/3tWdd8/cqIFfErOp8IEdnAQmJlhlaAMuI6tDAOCGWjSL/JUMQVIvkObWu2a33fm35OD5q+NyCBquM1a11i/M78oKq0xR8l02OcOKtCdPf09cUBwY8yiRRDIa2PVEpmzHcE5YMXY8IL
```
Comment (hiddeco) [inline test/known_hosts]: We probably want to perform the `ssh-keyscan` during the test as changing the `Dockerfile` changes the fingerprint iirc.

Comment (stefanprodan) [reply test/known_hosts]: I've deleted that file from the repo. The known_hosts file is now generated in CI and uploaded to artifacts.

File: test/ssh-keyscan.sh [hunk 2]
```diff
@@ -3,5 +3,5 @@
 repo_root=$(git rev-parse --show-toplevel)
 gen_dir=$(mktemp -d)
 
-kubectl exec -it deployment/gitsrv ssh-keyscan gitsrv > "$gen_dir/known_hosts"
-cat "$gen_dir/known_hosts" |  grep gitsrv
+kubectl exec -it deployment/gitsrv ssh-keyscan gitsrv > known_hosts.txt
+cat known_hosts.txt | grep gitsrv
```
Comment (hiddeco) [inline test/ssh-keyscan.sh]:
```suggestion
grep -E gitsrv known_hosts.txt
```

**Review by hiddeco — APPROVED** [review]
Drastic improvement! Thanks Stefan ✨

**Pull request merged**